libs/two/directory/src/lib/services/directory/directory.service.ts

```diff
@@ -0,0 +1,224 @@
+                        const workFile = `${this.workDir}\\${filename}`;
+                        const archFile = `${this.archiveDir}\\${filename}`;
+                        fs.copyFile(filename, workFile, (copyErr) => {
+                          if (copyErr) {
+                            throw copyErr;
+                          } else {
+                            // this.notifyValidationService(filename);
+                            fs.move(workFile, archFile, {
+                              overwrite: true
+                            })
+                              .then(() => {
+                                console.log('success');
+                              })
+                              .catch((moveErr) => {
+                                console.error(moveErr);
+                              });
+                          }
+                        });
+                        // fs.copyFileSync(filename, destFile);
+                      } else {
+                        console.log('Invalid filename...');
+                      }
+                    });
+                  } else {
+                    console.warn('Unsupported extension, ignoring...');
+                  }
+                });
+              } else {
+                console.warn('File is unaccessible, ignoring...');
+              }
+            });
+          }
+        });
+      }
+    });
+  }
+  /**
+   * Watches an array of files in the source directory
+   *
+   * @memberof DirectoryService
+   */
+  public async watchFilesForChange() {
+    const filePaths: string[] = [];
+    this.files.forEach((value, index) => {
+      filePaths.push(`${this.sourceDir}\\${value}`);
+    });
+
+    fs.access(this.sourceDir, fs.constants.F_OK, (err) => {
+      if (err) {
+        throw err;
+      }
+      try {
+        chokidar
+          .watch(filePaths, {
+            atomic: true,
+            depth: 0,
+            awaitWriteFinish: true
+          })
+          .on('change', (filePath: string, stats: Stats) => {
+            console.log(filePath);
+            const tokens = filePath.split('\\');
+            const fileName = tokens[tokens.length - 1];
+            const workFile = `${this.workDir}\\${fileName}`;
+            fs.copyFile(filePath, workFile, (copyErr) => {
+              if (copyErr) {
+                throw copyErr;
+              }
+              this.notifyValidationService(workFile);
+            });
+          });
+      } catch (error) {
+        throw error;
+      }
+    });
+  }
+
+  /**
+   * Checks the current file extension against a set list of acceptable extensions.
+   *
+   * @param {string} ext
+   * @returns {Promise<boolean>}
+   * @memberof DirectoryService
+   */
+  private async extIsValid(ext: string): Promise<boolean> {
+    const exts = ACCEPTABLE_EXTS;
+    return new Promise((resolve, reject) => {
+      const isValid = exts.includes(ext);
+      return resolve(isValid);
+    });
+  }
```

Comment:
Nothing here is asynchronous. Remove promise wrapper and async method keyword.

Reply:
`async` keyword wraps the return in a promise, but still nothing here is asynchronous and can be a straight synchronous return.

```js

private extIsValid(ext: string): boolean {
    const exts = ACCEPTABLE_EXTS;
    return new Promise((resolve, reject) => {
      const isValid = exts.includes(ext);
      return resolve(isValid);
    });
  }
```

And the affected code:

```js
                this.extIsValid(ext).then((validExt) => { // this line becomes unnecessary since you don't have to resolve a synchronous value
                  if (validExt) {

                  } else {
                    console.warn('Unsupported extension, ignoring...');
                  }
                }); // this line becomes unnecessary since you don't have to resolve a synchronous value
```